customization-base/src/main/java/com/azure/autorest/customization/PropertyCustomization.java

```diff
@@ -73,6 +73,108 @@ public PropertyCustomization rename(String newName) {
         return new PropertyCustomization(editor, languageClient, packageName, className, classSymbol, classCustomization, newName);
     }
 
+    /**
+     * Add an annotation to a property in the class.
+     *
+     * @param annotation the annotation to add. The leading @ can be omitted.
+     * @return the current property customization for chaining
+     */
+    public PropertyCustomization addAnnotation(String annotation) {
+        if (!annotation.startsWith("@")) {
+            annotation = "@" + annotation;
+        }
+
+        URI fileUri = classSymbol.getLocation().getUri();
+        Optional<SymbolInformation> symbol = languageClient.listDocumentSymbols(fileUri)
+                .stream().filter(si -> si.getName().equals(propertyName) && si.getKind() == SymbolKind.FIELD)
+                .findFirst();
+        if (symbol.isPresent()) {
+            int i = fileUri.toString().indexOf("src/main/java/");
+            String fileName = fileUri.toString().substring(i);
+            if (editor.getContents().containsKey(fileName)) {
+                int line = symbol.get().getLocation().getRange().getStart().getLine();
+                Position position = editor.insertBlankLine(fileName, line, true);
+                editor.replace(fileName, position, position, annotation);
+
+                FileEvent fileEvent = new FileEvent();
+                fileEvent.setUri(fileUri);
+                fileEvent.setType(FileChangeType.CHANGED);
+                languageClient.notifyWatchedFilesChanged(Collections.singletonList(fileEvent));
+
+                Optional<CodeAction> generateAccessors = languageClient.listCodeActions(fileUri, symbol.get().getLocation().getRange())
+                        .stream().filter(ca -> ca.getKind().equals(CodeActionKind.SOURCE_ORGANIZEIMPORTS.toString()))
+                        .findFirst();
+                if (generateAccessors.isPresent()) {
+                    WorkspaceEditCommand command;
+                    if (generateAccessors.get().getCommand() instanceof WorkspaceEditCommand) {
+                        command = (WorkspaceEditCommand) generateAccessors.get().getCommand();
+                        for (WorkspaceEdit workspaceEdit : command.getArguments()) {
+                            Utils.applyWorkspaceEdit(workspaceEdit, editor, languageClient);
+                        }
+                    }
+                }
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Remove an annotation from the property.
+     *
+     * @param annotation the annotation to remove from the property. The leading @ can be omitted.
+     * @return the current property customization for chaining
+     */
+    public PropertyCustomization removeAnnotation(String annotation) {
```

Comment:
Will this remove the annotation from the constructor too if the property had an annotation that was also used in the ctor args?

```java
    @JsonProperty(value = "length", required = true)
    private float length;

    @JsonCreator
    public Fish(@JsonProperty(value = "length", required = true) float length) {
        this.length = length;
    }
```

Reply:
The user will have to manually remove the annotation on the cotr. There may be future convenience customizations to "make a property optional".